src/m365/search/commands/externalconnection/externalconnection-add.spec.ts

```diff
@@ -0,0 +1,238 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import appInsights from '../../../../appInsights';
+import auth from '../../../../Auth';
+import { Logger } from '../../../../cli';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import commands from '../../commands';
+const command: Command = require('./externalconnection-add');
+
+describe(commands.EXTERNALCONNECTION_ADD, () => {
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  const externalConnectionAddResponse: any = {
+    "@odata.context": "https://graph.microsoft.com/beta/$metadata#connections/$entity",
+    "id": "TestConnectionForCLI",
+    "name": "Twitter Connector",
+    "description": "Connector for showing key tweets",
+    "connectorId": null,
+    "state": null,
+    "ingestedItemsCount": null,
+    "searchSettings": null,
+    "activitySettings": null,
+    "complianceSettings": null,
+    "configuration": {
+      "authorizedApps": [
+        "00000000-0000-0000-0000-000000000000"
+      ],
+      "authorizedAppIds": [
+        "00000000-0000-0000-0000-000000000000"
+      ]
+    }
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    (command as any).items = [];
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    Utils.restore([
+      request.post
+    ]);
+  });
+
+  after(() => {
+    Utils.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.EXTERNALCONNECTION_ADD), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('adds an external connection', (done) => {
+    const options: any = {
+      debug: false,
+      id: 'TestConnectionForCLI',
+      name: 'Test Connection for CLI',
+      description: 'Test connection that will not do anything'
+    };
+
+    command.action(logger, { options: options } as any, () => {
+      try {
+        assert(loggerLogSpy.calledWith(externalConnectionAddResponse));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+    done();
+  });
```

Comment:
This test was failing, if we want to verify the post response in the test then we should mock the post request and inspect the result.
 
```suggestion
  it('adds an external connection', (done) => {
    const postStub = sinon.stub(request, 'post').callsFake((opts) => {
      if (opts.url === `https://graph.microsoft.com/v1.0/external/connections`) {
        return Promise.resolve(externalConnectionAddResponse);
      }

      return Promise.reject('Invalid request');
    });

    const options: any = {
      debug: false,
      id: 'TestConnectionForCLI',
      name: 'Test Connection for CLI',
      description: 'Test connection that will not do anything'
    };

    command.action(logger, { options: options } as any, () => {
      try {
        assert.strictEqual(postStub.getCall(0).args[0].data, externalConnectionAddResponse);
        done();
      }
      catch (e) {
        done(e);
      }
    });
  });
```